src/runtime-rs/crates/resource/src/cgroups/mod.rs

```diff
@@ -109,7 +113,22 @@ impl CgroupsResource {
     /// overhead_cgroup_manager to the parent and then delete the cgroups.
     pub async fn delete(&self) -> Result<()> {
         for cg_pid in self.cgroup_manager.tasks() {
-            self.cgroup_manager.remove_task(cg_pid)?;
+            // For now, we can't guarantee that the thread in cgroup_manager does still
+            // exist. Once it exit, we should ignor that error returned by remove_task
+            // to let it go.
+            if let Err(error) = self.cgroup_manager.remove_task(cg_pid) {
```

Comment:
It is suggested that `remove_task_by_tgid` should be used instead of `remove_task` to adapt to cgroup v2. Cgroup v2 currently only supports deleting tasks in `thread mode`.

In order to shield the caller from judging the cgroup mode, https://github.com/kata-containers/cgroups-rs/pull/104 has been modified.

Reply:
OK, thanks @yaoyinnan

Reply:
Hi @yaoyinnan , as we discussed earlier, `remove_task_by_tgid` may move redundant tasks to root cgroup, and this is not what we expected. Ignoring this error message maybe more reasonable.